modules/reactor/src/meshgenerators/PatternedCartesianMeshGenerator.C

```diff
@@ -0,0 +1,753 @@
+  const Real extra_dist_shift_0 = _pattern_boundary == "expanded" ? pitch_array.front() / 5.0 : 0.0;
+  std::vector<unsigned int> peripheral_duct_intervals;
+  if (_pattern_boundary == "expanded")
+  {
+    if (_has_assembly_duct)
+      for (unsigned int i = 0; i < _duct_sizes.size(); i++)
+      {
+        if (_duct_sizes_style == PolygonSizeStyle::radius)
+          _duct_sizes[i] *= std::cos(M_PI / 4.0);
+        /// As square geometry is used here, size of patterned mesh can be straightforwardly calculated.
+        extra_dist.push_back(0.5 * (_duct_sizes[i] * 2.0 - pitch_array.front() * _pattern.size()));
+        peripheral_duct_intervals.push_back(_duct_intervals[i]);
+      }
+    // calculate the distance between the larger square boundary and the boundary of stitched unit
+    // squares this is used to decide whether deformation is needed when cut-off happens or when
+    // the distance is small.
+    extra_dist.push_back(0.5 * (_pattern_pitch - pitch_array.front() * _pattern.size()));
+    peripheral_duct_intervals.insert(peripheral_duct_intervals.begin(), _background_intervals);
+
+    // In same cases, when the external square size is small enough, the external square boundary
+    // may either be very close to the input square meshes that are near the boundary or even cut
+    // off the these squares. As long as the ring regions are not cut off, the input squares can
+    // be deformed to accomodate the external square shape. This block sets up the range of mesh
+    // region that needs to be deformed.
+    if (extra_dist.front() <= extra_dist_tol)
+    {
+      extra_dist_shift = extra_dist_shift_0 - extra_dist.front();
+      for (Real & d : extra_dist)
+        d += extra_dist_shift;
+      y_min = _deform_non_circular_region
+                  ? max_radius_global // Currently use this, ideally this should be the max of the
+                                      // outer layer radii
+                  : (pitch_array.front() / 2.0);
+      y_max_0 = pitch_array.front() / 2.0 + extra_dist.front();
+      y_max_n = y_max_0 - extra_dist_shift;
+      if (y_max_n <= y_min)
+        mooseError("In PatternedCartesianMeshGenerator ",
+                   _name,
+                   ": the assembly is cut off so much that the internal structure that should not "
+                   "be altered is compromised.");
+    }
+  }
+
+  _pattern_pitch_meta = _pattern_pitch;
+
+  // int x_mov = 0;
+
+  const Real input_pitch((_pattern_boundary == "expanded" || !_generate_core_metadata)
+                             ? pitch_array.front()
+                             : _pattern_pitch);
+  std::vector<Real> control_drum_positions_x;
+  std::vector<Real> control_drum_positions_y;
+  std::vector<std::vector<Real>> control_drum_azimuthals;
+
+  std::unique_ptr<ReplicatedMesh> out_mesh;
+
+  for (unsigned i = 0; i < _pattern.size(); i++)
+  {
+    // Real deltax = (Real)(_pattern.size() - 1) * input_pitch / 2.0;
+    const Real deltax = 0.0;
```

Comment:
why 0 and not j * input_pitch ?

Reply:
This is just the initial shift. `j * input_pitch` is used in the j-loop.

Reply:
yeah it's added to deltax. 
I think it d be expected to see j*input_pitch here